pages/books.tsx

```diff
@@ -165,36 +196,46 @@ const BooksCatalog = (
     });
   }, [data, searchValue]);
   return (
-    <Layout>
-      <div id="bookCatalog">
-        <Box
-          sx={{
-            display: 'flex',
-            flexFlow: 'row nowrap',
-            justifyContent: 'space-between'
-          }}
-          className="bookCatalog-topbar"
-        >
-          <div className="bookCatalog-topbar-search">
-            <Search search={searchValue} setSearch={setSearch} />
-          </div>
-          <div className="bookCatalog-topbar-buttons">
-            <AddBookModal bookData={data} />
-          </div>
-        </Box>
-        <Table rows={filteredItems || []} columns={tableColumns} />
-        <Box
-          className="bookCatalog-checkButtons"
-          sx={{
-            display: 'flex',
-            flexFlow: 'row-reverse nowrap'
-          }}
-        >
-          <CheckInOrOut title="Check In" CheckInOrOut="Check In" />
-          <CheckInOrOut title="Check Out" CheckInOrOut="Check Out" />
-        </Box>
-      </div>
-    </Layout>
+    <BooksContext.Provider value={{ data: data, setData: setData }}>
```

Comment:
Full disclosure - Never used a context provider before so can't offer any suggestions on this

Reply:
useContext() hook seemed like the easiest way to give all child components of books.tsx access to the books table data, so it can update the table on the frontend whenever API calls are made.